channel.go

```diff
@@ -184,10 +184,22 @@ func (ch *Channel) Delete() error {
 }
 
 // Truncate removes all messages from the channel.
-func (ch *Channel) Truncate() error {
+// You can pass in options such as hard_delete, skip_push
+// or a custom message.
+func (ch *Channel) Truncate(options ...map[string]interface{}) error {
```

Comment:
I used the `...` operator to keep backward compatibility. But now we need to check for the length of the array...

Any better ideas?

Reply:
Define a struct to hold final options and accept a slice option function. Then, initialize final struct with defaults and apply all given option functions.

```go
type truncateOptions struct {
  HardDelete bool
  TruncatedAt time.Time
}

type TruncateOption func(*truncateOptions)

func WithTruncateHardDelete(hardDelete bool) {
  return func(f *finalResult) {
    f.HardDelete = hardDelete
  }
}

func (ch *Channel) Truncate(options ...TruncateOption) error {
  result := &truncateOptions{}
  for _, op := range options {
    op(result)
  }
  // use result as you wish
}

// client
ch.Truncate(
	WithTruncateHardDelete(true),
        WithTruncateAt(oneDayAgo))
```

Reply:
interesting. I can make it work, although might be a bit complex for such a simple feature :)